lib/downloader.rb

```diff
@@ -26,11 +27,33 @@ def setTermSize
 end
 
 def downloader (*args)
+  # downloader: wrapper for all Chromebrew downloaders (`net/http`,`curl`...)
+  # Usage: downloader <url>, <filename::optional>, <retry_count::optional>,
+  #                   <verbose::optional>, sha256sum: <optional>
+  #
+  #           <url>: URL that points to the target file
+  #      <filename>: (Optional) Output path/filename
+  #   <retry_count>: (Optional) Maximum retry times, default: 3
+  #       <verbose>: (Optional) Verbose output
+  #     <sha256sum>: (Optional) SHA256 checksum, verify downloaded file with given checksum (if provided)
+  #
   setTermSize
   # reset width settings after terminal resized
   trap('WINCH') { setTermSize }
 
   uri = URI(args[0]) # read url from given params
+  filename = args[1] || File.basename(uri.path)
+
+  # verify with given checksum if checksum is provided in optional args (`downloader "url", sha256sum: "..."`)
+  if args[-1].is_a?(Hash) and args[-1].has_key?(:sha256sum)
+    check_sha256 = true
+    sha256sum = args[-1][:sha256sum]
+  else
+    check_sha256 = false
```

Comment:
Shouldn't we do a sha256sum check on everything?  Throw an error instead?

Reply:
The real check is in lines 80-82

Reply:
Right, but if `check_sha256` is false, those lines are skipped.

Reply:
Yeah, if `sha256sum` is not provided, then skip those checks

Reply:
I don't think you understand what I'm suggesting.  If those checks are skipped then there is no way to verify if the download is corrupt.

Reply:
I mean if the `sha256sum` parameter is provided (should be provided by `crew` is sha256sum != `SKIP`), then `check_sha256` will be set to `true`, then the checksum check will be performed.

Reply:
(we don't have any way to verify the file if the `sha256sum` param is not provided :) )

Reply:
Right, which is why I'm suggesting we make that a required parameter.  If the parameter is not provided then throw an error.

Reply:
The reason that I make it an optional parameter is for external use (like using for downloading in command line), and it takes lots of work to add a required parameter between the `<url>` and `<verbose>` arg :)

Reply:
Don't we already have a `crew download` command?  I'm just wondering about package usage.  I'm not sure we should be using it on the command line because the use case is to check if files are downloaded correctly.

Reply:
Changed `sha256sum` to a required parameter (it is much simpler than I think...)